src/LuaTransformer.ts

```diff
@@ -2697,7 +2715,13 @@ export class LuaTransformer {
             case ts.SyntaxKind.EqualsEqualsEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsToken:
             case ts.SyntaxKind.ExclamationEqualsEqualsToken:
-                return this.transformBinaryOperation(lhs, rhs, expression.operatorToken.kind, expression);
+                // Custom handling for 'typeof(foo) === "type"'
+                if (ts.isTypeOfExpression(expression.left) && tstl.isStringLiteral(rhs)) {
```

Comment:
I don't like adding extra logic here, we could just add this to transformBinaryOperation right?

Reply:
I agree having the logic out here is not very nice. But, I need to check the original ts expression types, which are not available in `transformBinaryOperation` currently. I also need to limit this logic to comparison operators, so I would need to re-check the operator in that method.

Reply:
I've refactored things a bit to clean it up.